search/includes/classes/class-search.php

```diff
@@ -237,4 +240,44 @@ static function ep_skip_query_integration( $skip ) {
 		// The filter is checking if we should _skip_ query integration
 		return ! ( $query_integration_enabled || $query_integration_enabled_legacy );
 	}
+
+	/**
+	 * Filter for ep_pre_request_host
+	 *
+	 * Return the next host in our enpoint list if it's defined. Otherwise, return the last host.
+	 */
+	static function filter__ep_pre_request_host( $host, $failures, $path, $args ) {
+		if( ! defined( 'VIP_ELASTICSEARCH_ENDPOINTS' ) ) { 
+			return $host;
+		}
+
+		return self::round_robin_hosts( $host, VIP_ELASTICSEARCH_ENDPOINTS );
+	}
+
+	/**
+	 * Return the next host in the list based on the current host
+	 */
+	static function round_robin_hosts( $host, $hostList ) {
+		$cur_index = array_search( $host, $hostList );
+		$max_index = count( $hostList ) - 1;
+
+		if ( $cur_index === $max_index or is_null( $cur_index ) ) {
+			return $hostList[ 0 ];
+		}
+
+		return $hostList[ $cur_index + 1 ];
+	} 
+
+	/**
+	 * Given a list of endpoints, randomly select one for load balancing purposes.
+	 */
+	static function load_balance_endpoints( $endpoints ) {
```

Comment:
Similar note - this could be called something like this, which indicates what the function actually does (it gets a random host from the list)

```suggestion
	static function get_random_host( $endpoints ) {
```